mino/minogrpc/server.go

```diff
@@ -45,6 +47,83 @@ type overlayServer struct {
 	closer   *sync.WaitGroup
 }
 
+func (o overlayServer) Join(ctx context.Context, req *JoinRequest) (*JoinResponse, error) {
+	// 1. Check validity of the token.
+	if !o.tokens.Verify(req.Token) {
+		return nil, xerrors.Errorf("token '%s' is invalid", req.Token)
+	}
+
+	// 2. Share certificates to current participants.
+	list := make(map[mino.Address][]byte)
+	o.certs.Range(func(addr mino.Address, cert *tls.Certificate) bool {
+		list[addr] = cert.Leaf.Raw
+		return true
+	})
+
+	peers := make([]*Certificate, 0, len(list))
+	res := make(chan error, 1)
+
+	for to, cert := range list {
+		text, err := to.MarshalText()
+		if err != nil {
+			return nil, xerrors.Errorf("couldn't marshal address: %v", err)
+		}
+
+		msg := &Certificate{Address: text, Value: cert}
+
+		// Prepare the list of send back to the new node.
```

Comment:
it's too late for me to rephrase it, but it seems that a word should be changed.